programs/solana/packages/solana-ibc-types/src/router.rs

```diff
@@ -50,6 +50,23 @@ impl Packet {
     }
 }
 
+/// Payload metadata for chunked operations
+#[derive(AnchorSerialize, AnchorDeserialize, Clone, Debug)]
+pub struct PayloadMetadata {
+    pub source_port: String,
+    pub dest_port: String,
+    pub version: String,
+    pub encoding: String,
+    pub total_chunks: u8,
+}
+
+/// Proof metadata for chunked operations
+#[derive(AnchorSerialize, AnchorDeserialize, Clone, Debug)]
+pub struct ProofMetadata {
+    pub height: u64,
+    pub total_chunks: u8,
+}
+
```

Comment:
I don't see how one would submit a non-chunked payload.

Reply:
You add it inline to the packet